lib/cocoapods/external_sources/abstract_external_source.rb

```diff
@@ -111,11 +111,15 @@ def pre_download(sandbox)
         title = "Pre-downloading: `#{name}` #{description}"
         UI.titled_section(title,  :verbose_prefix => '-> ') do
           target = sandbox.pod_dir(name)
-          download_result = Downloader.download(download_request, target, :can_cache => can_cache)
+          begin
+            download_result = Downloader.download(download_request, target, :can_cache => can_cache)
+          rescue Pod::DSLError => e
+            raise Informative, "Failed to load '#{name}' podspec: #{e.message}"
```

Comment:
Can you add this `begin` `rescue` to `store_podspec` in this file too? That'll handle similar cases for `PathSource` and `PodspecSource`.